### ISSUE TITLE
Add a YAML metadata parser

### ISSUE THREAD

**Issue by alfredbaudisch** [alfredbaudisch/pardall_markdown#41]

**Comment by alfredbaudisch** [comment]
@rockchalkwushock Wanna tackle this one? I added the steps to the issue.

**Comment by rockchalkwushock** [comment]
@alfredbaudisch yep I think I can tackle this pretty easily. I will review it more this afternoon and if I have any questions post them here.
